File: backend/backend/src/tweets/response/TweetResponse.py
from pydantic import BaseModel, Field, field_validator
from typing import List, Optional, Literal
from datetime import datetime


class CommentResponse(BaseModel):
    id: int
    tweet_id: int
    user_id: str
    text: str
    parent_comment_id: Optional[int]
    like_count: int
    is_liked: bool
    created_at: datetime
    edited_at: Optional[datetime] = None
    user_name: Optional[str] = None
    photo: Optional[str] = None
    is_organizational: bool = False
    is_prime: bool = False


class TweetMediaResponse(BaseModel):
    media_type: Literal["image/jpeg", "image/png", "image/jpg", "image/webp"]
    media_path: Optional[str] = Field(None, description="Path to media file on server")


class TweetResponse(BaseModel):
    id: int
    user_id: str
    text: str
    media: List[TweetMediaResponse]
    view_count: int
    like_count: int
    comment_count: int
    share_count: int
    bookmark_count: int
    is_shared: bool
    is_liked: bool
    is_bookmarked: bool
    created_at: datetime
    edited_at: datetime | None
    user_name: Optional[str] = None
    photo: Optional[str] = None
    is_organizational: bool
    is_prime: bool
    comments: List[CommentResponse]

File: frontend/app/src/main/java/com/app/cirkle/data/model/tweets/response/TweetResponse.kt
package com.app.cirkle.data.model.tweets.response



import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.app.cirkle.data.model.common.MediaResponse
import com.app.cirkle.domain.model.tweet.Media

@JsonClass(generateAdapter = true)
data class TweetResponse(
    @Json(name = "id") val id: Long,
    @Json(name = "user_id") val userId: String,
    @Json(name = "text") val text: String,
    @Json(name = "media") val media: List<MediaResponse>,
    @Json(name = "view_count") val viewCount: Int,
    @Json(name = "like_count") val likeCount: Int,
    @Json(name = "comment_count") val commentCount: Int,
    @Json(name = "share_count") val shareCount: Int,
    @Json(name = "bookmark_count") val bookmarkCount: Int,
    @Json(name = "is_shared") val isShared: Boolean,
    @Json(name = "is_liked") val isLiked: Boolean,
    @Json(name = "is_bookmarked") val isBookmarked: Boolean,
    @Json(name = "created_at") val createdAt: String,
    @Json(name = "edited_at") val editedAt: String?,
    @Json(name = "user_name") val userName: String,
    @Json(name = "photo") val photoUrl: String?,
    @Json(name = "is_organizational") val isOrganizational: Boolean,
    @Json(name = "is_prime") val isPrime: Boolean,
    @Json(name = "comments") val comments: List<CommentResponse>
)



